lib/igniter/project/application.ex

```diff
@@ -16,10 +16,8 @@ defmodule Igniter.Project.Application do
   @doc "Adds a new child to the `children` list in the application file"
   @spec add_new_child(Igniter.t(), module() | {module, term()}) :: Igniter.t()
   def add_new_child(igniter, to_supervise) do
-    project = Mix.Project.get!()
-
     to_perform =
-      case project.application()[:mod] do
+      case mod_definition(igniter) do
         nil -> {:create_an_app, Igniter.Code.Module.module_name("Application")}
```

Comment:
We need to have an `:error` case here, where we couldn't parse it. Then we'd want to prompt the user.
Something like:

```
Could not parse project configuration. Do we need to create an application? y/n
# if no
What is the name of the application?
```